src/app/pages/psbt-request/components/psbt-decoded-request/psbt-decoded-request-views/match-inputs-and-outputs.ts

```diff
@@ -0,0 +1,92 @@
+import * as btc from '@scure/btc-signer';
+
+import { isDefined } from '@shared/utils';
+
+import { PsbtInputForUi } from './use-psbt-decoded-request';
+
+export interface InputOutputPair {
+  inputs: PsbtInputForUi[];
+  output: btc.TransactionOutputRequired;
+}
+
+function consolidateInputsWithSameAddress(inputs: PsbtInputForUi[]): PsbtInputForUi[] {
+  const utxo: Record<string, PsbtInputForUi> = {};
+  inputs.forEach(({ address, value, ...rest }) => {
+    if (isDefined(utxo[address])) {
+      utxo[address] = {
+        ...utxo[address],
+        value: utxo[address].value + value,
+      };
+    } else {
+      utxo[address] = { address, value, ...rest };
+    }
+  });
+  return Object.values(utxo);
+}
+
+interface MatchInputsOutputsArgs {
+  psbtInputs: PsbtInputForUi[];
+  unsignedOutputs: btc.TransactionOutputRequired[];
+}
+export function matchInputsOutputs({ psbtInputs, unsignedOutputs }: MatchInputsOutputsArgs): {
+  fee: number;
+  inputOutputPairs: InputOutputPair[];
+} {
+  const pairs: InputOutputPair[] = [];
+  const availableInputs = [...psbtInputs];
+  let remainingInputValue = 0;
+
+  for (const output of unsignedOutputs) {
+    const inputOutputPair: InputOutputPair = {
+      inputs: [],
+      output: output,
+    };
+
+    // Calculate the remaining value needed for this output
+    let remainingOutputValue = Number(output.amount);
+
+    // Iterate through the available inputs and add them until the output is fulfilled
+    for (let i = 0; i < availableInputs.length; i++) {
+      const input = availableInputs[i];
+
+      if (remainingInputValue === 0) remainingInputValue = input.value;
+
+      // Handle if the unsigned input at index has an inscription
+      if (availableInputs[i].unsignedUtxo?.inscriptions) {
+        // Make sure this is an inscription transfer value
+        if (remainingInputValue === 10000 || remainingOutputValue === 10000) {
```

Comment:
hm, why this value is `10000`?

Reply:
This is the transfer value of an inscription.

Reply:
may be create variable for it?

Reply:
would flag this as magic number, can we make a const?

Also, _is_ it the case an inscription is always 10k? It doesn't have to be. Is that what gamma does?

Reply:
> It is always 10000 for the sell or buy

Requests from Gamma? What if another marketplace tries to use PSBTs for inscriptions, and doesn't follow this rule? Main thinking here is that our implementation should try to avoid details specific to given apps.

Are there any ways to avoid this? Do we absolutely need this check?

Reply:
I thought this was an inscription rule? Magic Eden does same. I thought this was built into the protocol for some reason?

Reply:
Anyway, I will remove it and think of something else.